### ISSUE TITLE
Create only one target

### ISSUE THREAD

**Issue by autoantwort** [libuv/libuv#4086]
Currently different targets are created for static and dynamic build which makes it hard to consume libuv. Imho a lib should be able to consume libuv while not caring how it was build. 
In vcpkg every project that uses the targets provided by the config files must be patched because they hardcode the dynamic or static target: https://github.com/search?q=repo%3Amicrosoft%2Fvcpkg%20libuv%3A%3A&type=code

There are also different pkg-config files depending on libuv is build static or dynamic. Same Problem. 
I currently working on wpilib in vcpkg. There are also using `libuv`, but if libuv is build static it should be `libuv-static`. 
https://github.com/wpilibsuite/allwpilib/blob/d37f990ce3a8a36bc791dd989dcea2868759e556/CMakeLists.txt#L170-L171
I can now dynamically patch that so that it is replaced depending on how libuv is build, but this is ugly. 

So I would propose that you provide only one target `libuv` regardless if libuv was build static or dynamic. 
If people want to use dynamic and static version of libuv at the same time they could use a solution like in https://github.com/redis/hiredis/pull/1160#discussion_r1103806804.

**Comment by bnoordhuis** [comment]
I agree that, in retrospect, it would have been better to honor BUILD_SHARED_LIBS. I picked the two-outputs approach because building+testing twice would have been super awkward to set up on our CI back then.

We probably can't move wholesale to a single-output build in a single release because that breaks downstream users of the cmake build. There should be a transition period to give users time to upgrade, where the old and the new way both work.

Do you have suggestions on how to tackle that?

**Comment by autoantwort** [comment]
The first thing we should do is to set the goals. 
1. Do we want to provide two targets simultaneously (static/dynamic) when used via git submodule + `add_subdirectory`? 
2. Should it be possible to use a static and a dynamic variant of libuv via `find_package` simultaneously? 
3. Should we add a option to keep the current target names? 
4. Should it be possible to build a static and dynamic variant of libuv simultaneously? 

My personal opinion: 
1. Yes, otherwise this change will not be merged because it breaks peoples build. 
2. Same as 1. 
3. Same as 1. 
4. No, because this simplify the build scripts. 
The new default behavior should be: Respect `BUILD_SHARED_LIBS` and only generate the target `uv`.

**Comment by autoantwort** [comment]
I will create a PR for that :)

**Comment by vtjnash** [comment]
4. For the ability for distros to package libuv, I don't think you can remove the functionality of making both static and dynamic versions together.

**Comment by autoantwort** [comment]
Is running configure + build for static and dynamic not an option?

**Comment by vtjnash** [comment]
The expectation is that they can be installed side-by-side, not that the build itself is an issue

**Comment by autoantwort** [comment]
Yeah that should be supported. What I means was _*build* a static and dynamic variant of libuv simultaneously_ not _install a static and dynamic variant of libuv simultaneously_

**Comment by vtjnash** [comment]
How do you propose to install them if you cannot build them?

**Comment by autoantwort** [comment]
Like this:
```bash
cmake -B build -DBUILD_SHARED_LIBS=OFF
cmake --build build
cmake --install build
cmake -B build -DBUILD_SHARED_LIBS=ON
cmake --build build
cmake --install build
```
